### DISCUSSION TITLE
How to cleanly abort a Monitor?

### DISCUSSION THREAD

**Discussion by wsw70** [jamesoff/simplemonitor#1386] [General]
I am running the alpha version of my remote monitor (agentless ssh-based monitor) and at some point I wanted to abort the Monitor without returning neither a success nor a failure.  This happens in `run_test()` but could also be in `__init__()`.

Right now I am logging the error and raising an exception but this is not really clean (the user already knows about the problem and the Traceback does not add anything).

Is there a "proper" way to abort, or should I just continue raising the exception?

**Comment by jamesoff** [comment]
If you encounter a problem during `__init__()`, raise an exception which will cause simplemonitor to abort loading and report an error. This should be for invalid configurations or missing libraries, that kind of thing.

If something goes wrong during the running of the test, ideally you should `report_fail()` as normal and explain why in the message. An uncaught exception fron `run_test()` will be handled as a failure by the main loop, but handling it inside the monitor is better I think (because you have more control over how it's reported).

Hope that helps :)

**Comment by wsw70** [comment]
ok cool, understood
